app/scripts/modules/core/cluster/filter/clusterFilter.component.ts

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+import {CLUSTER_FILTER_SERVICE} from 'core/cluster/filter/clusterFilter.service';
+
+import {compact, uniq, map} from 'lodash';
+import {IScope, module} from 'angular';
+import {Application} from '../../application/application.model';
+
+export const CLUSTER_FILTER = 'cluster';
+
+let ngmodule = module(CLUSTER_FILTER, [
+  require('./collapsibleFilterSection.directive.js'),
+  CLUSTER_FILTER_SERVICE,
+  require('./clusterFilter.model.js'),
+  require('../../filterModel/dependentFilter/dependentFilter.service.js'),
+  require('./clusterDependentFilterHelper.service.js'),
+]);
+
+class ClusterFilterCtrl {
+  public categoryHeadings: any;
+  public stackHeadings: string[];
+  public instanceTypeHeadings: any;
+  public regionHeadings: any;
+  public accountHeadings: any;
+  public providerTypeHeadings: any;
+  public availabilityZoneHeadings: any;
+  public app: Application;
+  public tags: any[];
+
+  constructor(public $scope: IScope,
+              public clusterFilterService: any,
+              public ClusterFilterModel: any,
+              public $rootScope: IScope,
+              public clusterDependentFilterHelper: any,
+              public dependentFilterService: any,
+  ) {
+  }
+
+  public $onInit() {
+    const { $scope, $rootScope, ClusterFilterModel, clusterFilterService, app } = this;
```

Comment:
Yes it's destructuring to avoid `this.` everywhere.  It effectively transpiles to: 

```
const $scope = this.$scope;
const $rootScope = this.$rootScope;
const ClusterFilterModel = this.ClusterFilterModule;
... etc 
```